src/couch_db_plugin.erl

```diff
@@ -0,0 +1,65 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_db_plugin).
+
+-export([
+    validate_dbname/2,
+    before_doc_update/2,
+    after_doc_read/2,
+    validate_docid/1,
+    check_is_admin/1
+]).
+
+-include_lib("couch/include/couch_db.hrl").
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+validate_dbname(DbName, Normalized) ->
+    Handle = couch_epi:get_handle(couch_db),
```

Comment:
In other apps, you defined `?SERVICE_ID` macro for handle name. Why not here as well?